modules/nf-core/bbmap/filterbyname/main.nf

```diff
@@ -0,0 +1,59 @@
+process BBMAP_FILTERBYNAME {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::bbmap=39.01" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bbmap:39.01--h311275f_0':
+        'quay.io/biocontainers/bbmap:39.01--h311275f_0' }"
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("*.$output_extension"), emit: reads
+    tuple val(meta), path('*.log')                , emit: log
+    path "versions.yml"                           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args               = task.ext.args   ?: [ args:"" ]
+    args.args = args.args ? "$args.args" : ""
+
+    def prefix             = task.ext.prefix ?: "${meta.id}"
+    def interleaved_output = args.interleaved_output ? true : false
+    def input  = meta.single_end ? "in=${reads}" :
+        "in=${reads[0]} in2=${reads[1]}"
+
+    def avail_mem = 3
+    if (!task.memory) {
+        log.info '[filterbyname] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
+    } else {
+        avail_mem = task.memory.giga
+    }
+
+    output_extension   = args.output_extension ? "$args.output_extension" :
```

Comment:
For the ouput, this can be handled via the modules.config and passed in via the prefix, for example how we deal with bam input but cram output in GATK4 Markduplicates

Reply:
either prefix can be evaluated with a closure for this or it can be fixed by the pipeline developer

Reply:
That's smart!

Reply:
@FriederikeHanssen I am not following how this would solve an issue of "unknown output format".

Reply:
Also, the way I interpret `prefix` - it is everything before extension. Controlling file extension through `prefix` is confusing for me.